app/components/autocomplete/at_mention/index.js

```diff
@@ -17,12 +18,23 @@ import {
 } from 'app/selectors/autocomplete';
 import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
 
+import {haveIChannelPermission} from 'mattermost-redux/selectors/entities/roles';
+import {Permissions} from 'mattermost-redux/constants';
+
 import AtMention from './at_mention';
 
 function mapStateToProps(state, ownProps) {
     const {cursorPosition, isSearch} = ownProps;
     const currentChannelId = getCurrentChannelId(state);
 
+    const useChannelMentions = !isMinimumServerVersion(state.entities.general.serverVersion, 5, 22) || haveIChannelPermission(
```

Comment:
I may be alone on this but I'm not a huge fan with the way this is done. At first glance, I thought `useChannelMentions` would happen if the minimum server version wasn't `5.22` but I realised you are inverting the `true` such that if it is , it goes on to check the channel permissions. It was a little confusing at first glance so do you think it would better if we just wrote it out using a if statement to make the code more readable?

```suggestion
    const useChannelMentions = !isMinimumServerVersion(state.entities.general.serverVersion, 5, 22) || haveIChannelPermission(
```
```suggestion
   let useChannelMentions;
    if (isMinimumServerVersion (...)) {
        useChannelMentions =haveIChannelPermission(...)
    }
```

Reply:
I have no strong opinions about not doing it this way - modified in my latest commit - also updated redux hash to make it testable locally without cloning redux changes too

Reply:
Thanks for the quick change! I think it would just be good to keep in mind to update the redux hash to the master after your redux changes are merged and before you push these changes to master.

Reply:
Scratch my last comment. I didn't pay attention to the redux changes having already been merged. Sorry!